### PR TITLE
Remove remaining master_create_distributed_table usages

### PR DESCRIPTION
**Pull request by marcocitus**
Made a pass on all the regression tests to remove the remaining master_create_distributed_table and master_create_worker_shards calls and removed their implementations.

### CODE REVIEW

File: src/test/regress/expected/multi_colocation_utils.out [hunk 1]
```diff
@@ -352,6 +322,7 @@ SELECT count(*) FROM pg_dist_partition WHERE colocationid IN (4, 5);
      0
 (1 row)
 
+DROP TABLE table1_group1, table2_group1, table3_group2, table4_group2, table5_groupX, table6_append;
```
Comment (marcocitus) [inline src/test/regress/expected/multi_colocation_utils.out:325]: This file was a bit messy due to the master_create_distributed_table giving invalid colocation ID, but dropping the tables seemed to give a decent balance between preserving coverage and minimizing test changes.

**Review by onderkalaci — APPROVED** [review]
Thanks for the clean-up! 

I think this PR should have DESCRIPTION?  Just in case there are people using `master_create_distributed_table ` and `master_create_worker_shards`?

**Comment by marcocitus** [comment]
> Just in case there are people using master_create_distributed_table and master_create_worker_shards?

Those were dropped in 10.0, they were recreated in tests.

**Comment by onderkalaci** [comment]
> Those were dropped in 10.0, they were recreated in tests.

Ah, I was confused as the UDFs are under `src/backend/..` but now I can follow

**Pull request merged**